mailbox/memory/src/test/java/org/apache/james/mailbox/inmemory/manager/InMemoryIntegrationResources.java

```diff
@@ -124,12 +126,12 @@ default RequireSearchIndex defaultMessageParser() {
         }
 
         interface RequireSearchIndex {
-            RequirePreDeletionHooks searchIndex(Function<MailboxManagerPreInstanciationStage, MessageSearchIndex> searchIndex);
+            RequirePreDeletionHooks searchIndex(Function<MailboxManagerSearchIndexStage, MessageSearchIndex> searchIndex);
```

Comment:
could also be step #13 a separated PR?

Reply:
It's **MUCH** more coupled to the content of this pull request.

Please spare me that.